tests/Feature/SearchConsoleReportTest.php

```diff
@@ -370,4 +370,47 @@ public function test_formats_daily_data_correctly(): void
             return true;
         });
     }
+
+    /**
+     * Test that the notification uses markdown template correctly.
+     */
+    public function test_notification_uses_markdown_template(): void
+    {
+        // Create mock report data
+        $reportData = [
+            'https://example.com/' => (object) [
+                'rows' => [
+                    (object) [
+                        'keys' => ['2024-01-01'],
+                        'clicks' => 100,
+                        'impressions' => 1000,
+                        'ctr' => 0.1,
+                        'position' => 5.5,
+                    ],
+                ],
+            ],
+        ];
+
+        // Create notification instance
+        $notification = new SearchConsoleReportNotification($reportData);
+
+        // Get the mail message
+        $mailMessage = $notification->toMail((object) []);
```

Comment:
MailMessage has the `render()` method, so you should be able to test the html converted from markdown.